src/qml/components/UploadProgressBar.qml

```diff
@@ -0,0 +1,115 @@
+import QtQuick 2.12
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import "../theme"
+
+
+Rectangle {
+    id: uploadProgressBar
+
+    property var model: models.fileCloudUploader
+    property int fileIndex: model.currentIndex
+    property int filesCount: model.fileNames.length
+    property real fileUploadValue: model.currentProcessedBytes * 100 / model.currentTotalBytes
+
+    property real uploadHeight: 60
+    property real uploadOpacity: 1
+
+    property real uploadFrom: 0
+    property real uploadTo: 100
+
+    color: Theme.mainBackgroundColor
+
+    Rectangle {
+        id: separator
+        anchors.left: parent.left
+        anchors.right: parent.right
+        height: 1
+        color: Theme.chatBackgroundColor
+    }
+
+    state: filesCount !== 0 ? "opened" : "closed"
+    states: [
+        State {
+            name: "opened"
+
+            PropertyChanges {
+                target: uploadProgressBar
+                opacity: uploadOpacity
+                height: uploadHeight
+            }
+        },
+        State {
+            name: "closed"
+
+            PropertyChanges {
+                target: uploadProgressBar
+                opacity: 0
+                height: 0
+            }
+        }
+    ]
+
+    transitions: [
+        Transition {
+            NumberAnimation {
+                properties: "opacity, height"
+                easing.type: Easing.InExpo
+                duration: Theme.animationDuration
+            }
+        }
+    ]
+
+    Column {
+        anchors {
+            left: parent.left
+            right: parent.right
+            leftMargin: Theme.margin
+            rightMargin: Theme.margin
+            verticalCenter: parent.verticalCenter
+        }
+        height: this.contentHeight
```

Comment:
Good practice to use "this", if something doesn't work without "this". Just leave contentHeight.
Optional fix

Reply:
+

Reply:
It doesn't work without this